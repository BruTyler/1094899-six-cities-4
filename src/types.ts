import {PlaceType} from './const';

export interface WithId<K=string | number> {
  id: K,
}

interface WithLocation {
  location: [number, number]
};

interface DifferentOptions {
  [key: string]: boolean | string | number
};

interface Host {
  avatar: string,
  name: string,
  isSuper: boolean,
}

interface SingleApartmentCard {
  cityId: string,
  type: PlaceType,
  description: string,
  rating: number,
  price: number,
  isPremium: boolean,
  isFavourite: boolean,
  photo: string,
  fullDescription: string,
  photoSet: string[],
  bedrooms: number,
  adultsMax: number,
  goods?: string[],
  host: Host,
}

export type SingleApartment = WithId<number> & WithLocation & SingleApartmentCard;

export type City = WithId<string> & WithLocation & { 
  defaultZoom: number 
};

export type AuthInfo = WithId<number> & DifferentOptions & {
  email: string,
};

export interface AuthData {
  login: string,
  password: string,
};

export type Review = WithId<number> & {
  authorName: string,
  authorAvatar: string,
  rating: number,
  opinion: string,
  publishDate: string,
}

export interface ReviewFormData {
  comment: string, 
  rating: number, 
  apartmentId: number,
}

  
