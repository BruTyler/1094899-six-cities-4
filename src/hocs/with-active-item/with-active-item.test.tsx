import * as React from 'react';
import * as renderer from 'react-test-renderer';
import withActiveItem from './with-active-item';
import {WithId} from '../../types';

interface Props {
  activeItem: WithId<number>;
  onItemSelect: (ObjWithId: WithId<number>) => void;
}

const MockComponent: React.FunctionComponent<Props> = (props: Props) => {
  const {activeItem, onItemSelect} = props;

  return (
    <div onClick={() => onItemSelect(activeItem)}>
      {activeItem.id}
    </div>
  );
};

const MockComponentWrapped = withActiveItem(MockComponent);

describe(`withActiveItem render suit`, () => {
  it(`withActiveItem is rendered correctly`, () => {
    const tree = renderer.create(
        <MockComponentWrapped
          activeItem={{id: 1}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
