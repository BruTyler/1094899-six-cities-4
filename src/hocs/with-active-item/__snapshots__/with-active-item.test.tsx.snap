// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveItem render suit withActiveItem is rendered correctly 1`] = `
<div
  onClick={[Function]}
>
  1
</div>
`;
